### PR TITLE
[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73685

Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_linear_prepack_fp16_numerics
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
great, thanks!

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -2880,6 +2880,18 @@ def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
         self.assertEqual(Y_fp32, Y_fp32_ref,
                          msg="torch.ops.quantized.fbgemm_linear_dynamic results are off")
 
+    @given(
+        input_channels=st.integers(16, 32),
+        output_channels=st.integers(4, 8),
+        exponent=st.integers(0, 8))
+    def test_pack_unpack_fp16(self, input_channels, output_channels, exponent):
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:2887]: actually maybe this could have a better name, like test_linear_prepack_fp16_numerics

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

**Pull request closed without merging**